generate_openj9_scc.sh

```diff
@@ -0,0 +1,295 @@
+IS_APT_ENV=false
+# Intitialise default / allowed sample app status
+RUN_ECLIPSE=false
+RUN_TOMCAT=true
+
+# Intitialise default / required packages status
+INSTALL_GTK=false
+INSTALL_XVFB=false
+
+# App download locations
+DOWNLOAD_PATH_ECLIPSE=/tmp/eclipse
+DOWNLOAD_PATH_TOMCAT=/tmp/tomcat
+
+# App installation locations
+INSTALL_PATH_ECLIPSE="$HOME"/eclipse-home
+INSTALL_PATH_TOMCAT="$HOME"/tomcat-home
+
+# function to install packages
+function install_packages() {
+    if [ "$IS_APT_ENV" == true ]; then
+        # update the repositories
+        apt update
+
+        # Set non-interactive frontend (to avoid `tzdata` config options selection)
+        APT_INSTALL_CMD="DEBIAN_FRONTEND=noninteractive"
+        APT_INSTALL_CMD="${APT_INSTALL_CMD} apt install -y"
+
+        if [ "$INSTALL_GTK" == true ]; then
+            # install gtk+3.0
+            APT_INSTALL_CMD="${APT_INSTALL_CMD} gtk+3.0"
+        fi
+
+        if [ "$INSTALL_XVFB" == true ]; then
+            # install xvfb
+            APT_INSTALL_CMD="${APT_INSTALL_CMD} xvfb"
+        fi
+
+        eval "$APT_INSTALL_CMD"
+    fi
+}
+
+# function to download the applications
+function download_and_install_artifacts() {
+
+    # check for eclipse
+    if [ "$RUN_ECLIPSE" == true ]; then
+        # Creating a temporary directory for eclipse download
+        mkdir -p "$DOWNLOAD_PATH_ECLIPSE" "$INSTALL_PATH_ECLIPSE"
+
+
+        # Downloading eclipse
+        wget --directory-prefix="$DOWNLOAD_PATH_ECLIPSE"  http://www.mirrorservice.org/sites/download.eclipse.org/eclipseMirror/technology/epp/downloads/release/2020-06/M3/eclipse-java-2020-06-M3-linux-gtk-x86_64.tar.gz
+
+        # Extracting eclipse
+        tar -xvzf "$DOWNLOAD_PATH_ECLIPSE"/eclipse-java-2020-06-M3-linux-gtk-x86_64.tar.gz -C "$INSTALL_PATH_ECLIPSE"
+
+        # Removing the tar file
+        rm -rf "$DOWNLOAD_PATH_ECLIPSE"
+    fi
+
+    # check for tomcat
+    if [ "$RUN_TOMCAT" == true ]; then
+        # Creating a temporary directory for tomcat download
+        mkdir -p "$DOWNLOAD_PATH_TOMCAT" "$INSTALL_PATH_TOMCAT"
+
+        # Downloading tomcat
+        wget --directory-prefix="$DOWNLOAD_PATH_TOMCAT" https://downloads.apache.org/tomcat/tomcat-9/v9.0.35/bin/apache-tomcat-9.0.35.tar.gz
+
+        # Extracting tomcat
+        tar -xvzf "$DOWNLOAD_PATH_TOMCAT"/apache-tomcat-9.0.35.tar.gz -C "$INSTALL_PATH_TOMCAT"
+
+        # Removing the tar file
+        rm -rf "$DOWNLOAD_PATH_TOMCAT"
+    fi
+
+}
+
+# dry run to right size the cache
+function dry_run() {
+    # Creating base layer first instead of running sample programs to generate SCC directly
+    java -Xshareclasses:name=dry_run_scc,cacheDir=/opt/java/.scc,bootClassesOnly,nonFatal,createLayer -Xscmx$SCC_SIZE -version
+
+    # Pointing cache for sample program runs
+    export OPENJ9_JAVA_OPTIONS="-Xshareclasses:name=dry_run_scc,cacheDir=/opt/java/.scc,bootClassesOnly,nonFatal"
+
+    # check if `xvfb` installed
+    if [ "$INSTALL_XVFB" == true ]; then
```

Comment:
Please add a function for this, something like `run_xvfb`

Reply:
created function `run_xvfb` for creating a virtual screen